provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,251 @@
+            callback(new errors.InvalidOperationError('registration already in progress.'));
+          },
+          disconnect : (callback) => {
+            this._cancelCurrentOperation((err) => {
+              this._fsm.transition('idle', callback, err);
+            });
+          }
+        },
+        waiting_to_poll: {
+          _onEnter: (callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_018: [ `register` shall poll for operation status every `operationStatusPollingInterval` milliseconds ] */
+            this._pollingTimer = setTimeout(() => {
+              this._fsm.transition('polling',  callback);
+            }, this._operationStatusPollingInterval);
+
+          },
+          register: (authorization, forceRegistration, body, callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_036: [ `register` shall call the `callback` with an `InvalidOperationError` if it is called while a previous registration is in progress. ] */
+            callback(new errors.InvalidOperationError('registration already in progress.'));
+          },
+          disconnect : (callback) => {
+            this._cancelCurrentOperation((err) => {
+              this._fsm.transition('idle', callback, err);
+            });
+          }
+        },
+        polling: {
+          _onEnter: (callback) => {
+            this._sendOperationStatusRequest((err, body) => {
+              if (this._registrationInProgress()) { // make sure we weren't cancelled before doing something with the response
+                if (err) {
+                  this._fsm.transition('idle', callback, err);
+                } else if (body.status.toLowerCase() === 'assigning') {
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_028: [ If the operation status response contains a success status code with a 'status' that is 'Assigning', `register` shall fire an `operationStatus` event with the deserialized body and continue polling. ] */
+                  this.emit('operationStatus', body);
+                  this._fsm.transition('waiting_to_poll', callback);
+                } else if (body.status.toLowerCase() === 'assigned') {
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_027: [ If the operation status response contains a success status code with a 'status' of 'Assigned', `register` shall stop polling and call the `callback` with `err` == null and the body containing the deserialized body. ] */
+                  this._fsm.transition('idle', callback, null, body);
+                } else {
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_030: [ If the operation status response has a success code with a 'status' that is any other value, `register` shall call the callback with a `SyntaxError` error and stop polling. ] */
+                  this._fsm.transition('idle', callback, new SyntaxError('badly formed registration response'));
+                }
```

Comment:
looks very similar to the initial registration call response processing - maybe refactor inside one function? also, consider using a switch statement on body.status.toLowerCase()? technically it doesn't matter much but I feel it makes the code slightly more readable. it's a personal opinion though...